src/main/java/org/cirdles/topsoil/app/TSVTable.java

```diff
@@ -41,8 +41,8 @@
  */
 public class TSVTable extends TableView<Entry> {
 
-    private Path savePath;
-    private Dataset dataset;
+    private Optional<DatasetResource> datasetResource = Optional.empty();
+    private Optional<Dataset> unsavedDataset = Optional.empty(); //Due to pasteFromClipboard
```

Comment:
Because you need a path to create a `DatasetResource`, however data from the clipboard are unsaved and thus do not have a path.
We could automatically save them when pasted, but I'm not convinced of the user-friendliness of such a solution. Also, we'd need to have a `DatasetResourceFactory` inside the TSVTable, but as it need the basePath to be created,  we'd have to add a dependency to `TopsoilMainWindow` to retrieve it.